spec/models/notification_spec.rb

```diff
@@ -452,7 +453,7 @@
         new_title = "hehehe hohoho!"
         article.update_attribute(:title, new_title)
 
-        perform_enqueued_jobs do
+        Sidekiq::Testing.inline! do
```

Comment:
I think the mishap comes from the fact that this expectation is inside the block, a much better version would be:

```ruby
      it "updates the notification with the new article title" do
        new_title = "hehehe hohoho!"
        article.update_attribute(:title, new_title)

        described_class.update_notifications(article, "Published")

        sidekiq_perform_enqueued_jobs

        expected_notification_article_title = user2.reload.notifications.last.json_data["article"]["title"]
        expect(expected_notification_article_title).to eq(new_title)
      end
```